### ISSUE TITLE
metrics='l1'  vs 'metric': 'l2' 

### ISSUE THREAD

**Issue by Sandy4321** [microsoft/LightGBM#2618]
it is not clear from documentation what is difference 
between metrics='l1' and 'metric': 'l2' 

For example if we need binary metric for classification with L1 regularization to get sparse features
what metric to set and how to control (make sure) that only chosen strong features are used for prediction?
 
Under sparse features features I mean that only small number of best features are take part in classification 
as in L1 regularization for logistic regression  
https://scikit-learn.org/stable/auto_examples/linear_model/plot_logistic_l1_l2_sparsity.html
https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html
penaltystr, ‘l1’, ‘l2’, ‘elasticnet’ or ‘none’, optional (default=’l2’)
Used to specify the norm used in the penalization. The ‘newton-cg’, ‘sag’ and ‘lbfgs’ solvers support only l2 penalties. ‘elasticnet’ is only supported by the ‘saga’ solver. If ‘none’ (not supported by the liblinear solver), no regularization is applied

reference:
https://github.com/microsoft/LightGBM/blob/3400e3899d1c45997337bdbd0c8a2f729e805cbc/tests/python_package_test/test_engine.py#L413-L415

code example
        params_with_metric = {'metric': 'l2', 'verbose': -1}
        lgb.cv(params_with_metric, lgb_train, num_boost_round=10, nfold=3, stratified=False, shuffle=False,
               metrics='l1', verbose_eval=False) 

PS
by the way how different objective and metric are
when objective is used 
and when metric is used 

is it possible not to set metric at all, for example in case metric is not used

code reference 
https://github.com/microsoft/LightGBM/blob/3400e3899d1c45997337bdbd0c8a2f729e805cbc/tests/python_package_test/test_engine.py#L280-L281
            'objective': 'multiclass',
            'metric': 'multi_logloss',

**Comment by StrikerRUS** [comment]
> it is not clear from documentation what is difference
between metrics='l1' and 'metric': 'l2'

Refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric.
`l1` is `mean_absolute_error` and `l2` is `mean_squared_error`.
They are regression metrics.

> is it possible not to set metric at all,

Yes. Again, refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric.
> "None" (string, not a None value) means that no metric will be registered, aliases: na, null, custom

**Comment by Sandy4321** [comment]
it is not clear from documentation what is difference
between metrics='l1' and 'metric': 'l2'

Refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric.
l1 is mean_absolute_error and l2 is mean_squared_error.
They are regression metrics.

**Comment by Sandy4321** [comment]
it is not clear from documentation what is difference
between metrics='l1' and 'metric': 'l2'

Refer to https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric.
l1 is mean_absolute_error and l2 is mean_squared_error.
They are regression metrics.

You did not mentioned that in first case it is single and in second case it is plural
And different usage in code
code example
params_with_metric = {'metric': 'l2', 'verbose': -1}
lgb.cv(params_with_metric, lgb_train, num_boost_round=10, nfold=3, stratified=False, shuffle=False,
metrics='l1', verbose_eval=False
It is the question

**Comment by StrikerRUS** [comment]
> And different usage in code
code example
params_with_metric = {'metric': 'l2', 'verbose': -1}
lgb.cv(params_with_metric, lgb_train, num_boost_round=10, nfold=3, stratified=False, shuffle=False,
metrics='l1', verbose_eval=False
It is the question

I think that documentation is quite clear about the case when you set metrics in both `params` and `metrics` argument:

https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.cv.html#lightgbm.cv
> metrics (string, list of strings or None, optional (default=None)) – Evaluation metrics to be monitored while CV. If not None, the metric in params will be overridden.
